### DISCUSSION TITLE
Fix chmod of *.php files to make them non-executable

### DISCUSSION THREAD

**Discussion by GromNaN** [PHP-CS-Fixer/PHP-CS-Fixer#7414] [Feature ideas]
This is a common problem on open-source projects, and it's hard to detect in code review in GitHub UI. Some people create `*.php` files by making them executable (chmod +x).

Would it be possible to add a rule in php-cs-fixer to automatically set file permissions to ensure files are not executable. In contrast, files starting with a shebang must be executable.

Git stores only the executable bit, no need to bother with group perm.

**Comment by Wirone** [comment]
Hi @GromNaN! Personally I believe it's outside of the context of CS tool, because it should be explicit decision which file should be executable. This should be done with `git update-index --chmod=+x <file>`, so executable bit is shared through Git and applied when file is checked out. During code review it's possible to spot it, because the information is displayed, e.g. in Gitlab it looks like:

<img width="388" alt="image" src="https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/600668/68040f3b-dd68-47c9-a291-1701fd31e77d">

But if you think such rule can be helpful, it's always possible to create custom rule and share it publicly 🙂.

**Comment by kubawerlos** [comment]
> This is a common problem on open-source projects, and it's hard to detect in code review

@GromNaN it is not hard, we do it in this very repository quite simply: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.41.1/.github/workflows/sca.yml#L62
